### PR TITLE
#166265846 Send welcome email to user on signup

### PR DESCRIPTION
**Pull request by OvieMudi**
#### What does this PR do?
Adds feature where email is sent to user on signup

#### Description of Task to be completed?
After successful signup, ensure an email sent to provided receiver email

#### How should this be manually tested?
To test manually on your local computer, _run the following commands_.

```
# Clone this repository
$ git clone https://github.com/emp-daisy/HelloBooksLib.git

# Navigate to the application folder
$ cd HelloBooksLib

# Checkout the branch
$ git checkout ft/166265846/send-email-to-user-after-signup

# Install dependencies
$ npm install

# Start the app in development mode
$ npm run dev
```

- make a GET request to  ` localhost:3000/api/v1/auth/signup`
- sign up with a valid email
- ensure an email is sent to the signup email

NB: see screenshot below on how the layout should look
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166265846](https://www.pivotaltracker.com/story/show/166265846)

#### Screenshots
send_email1

### CODE REVIEW

**Review by encodedBicoding — DISMISSED** [review]

**Review by encodedBicoding — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -5,10 +5,10 @@
   "main": "index.js",
   "scripts": {
     "start": "node build/index.js",
-    "pretest": "export NODE_ENV=test && npm run migrate:undo && npm run migrate",
-    "test": "export NODE_ENV=test && jest --forceExit --verbose",
+    "pretest": "cross-env NODE_ENV=test npm run migrate:undo && cross-env NODE_ENV=test npm run migrate",
```
Comment (chuxmykel) [inline package.json:8]: Nice one. I see why this guy wasn't working now. Good job

File: server/controllers/userController.js [hunk 2]
```diff
@@ -28,6 +29,8 @@ const UserController = {
             const token = auth.generateToken(user);
 
               models.User.create(user).then((createdUser) => {
```
Comment (chuxmykel) [inline server/controllers/userController.js]: I thought we agreed to use async/await??

**Review by chuxmykel — CHANGES_REQUESTED** [review]

File: server/controllers/userController.js (hunk 2 above)
Comment (OvieMudi) [reply server/controllers/userController.js]: async/await will require me to use try/catch just to access & log errors. I thought it's an overkill. Do you recommend I change it?

Comment (encodedBicoding) [reply server/controllers/userController.js]: I actually did mine this way you did your initially, i felt the same thing too.

Comment (cvjude) [reply server/controllers/userController.js]: I think we should have the same convention
Secondly you could wrap the whole function in an async await

Comment (encodedBicoding) [reply server/controllers/userController.js]: Okay Ovie, please adjust this, so we can merge

Comment (cvjude) [reply server/controllers/userController.js]: Ovie also has to do a rebase to clear all the conflicts on the base branch

Comment (encodedBicoding) [reply server/controllers/userController.js]: Jude i just noticed that there is a .vscode folder that vame with your PR, kindle ignore it

File: server/helpers/mailer.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+class Mailer {
+  /**
+   * Email trasporter
+   * @param {String} to - Reciever email
+   * @param {String} from - Sender email
+   * @param {String} subject - Email subject
+   * @param {String} html - Email body
+   * @returns  {Object} - Mailer response
+   */
+  static sendMail(to, from, subject, html) {
```
Comment (OvieMudi) [inline server/helpers/mailer.js]: We're supposed to convert this to async/await method.
Then we can convert the `.then` to `try/catch`

File: server/controllers/userController.js [hunk 4]
```diff
@@ -26,30 +22,25 @@ const UserController = {
...
-              .catch((error) => {
-                return res.status(500).json({
-                    status: res.statusCode,
-                    error: 'Internal error',
-                  });
-              });
-            })
+            
+            try {
+                const USER = await models.Users.create(user);
```
Comment (chuxmykel) [inline server/controllers/userController.js]: USER or user?

File: server/controllers/userController.js [hunk 5]
```diff
@@ -26,30 +22,25 @@ const UserController = {
...
+                const data = {
+                    token,
+                    id: USER.id,
+                    firstName: USER.firstName,
+                    lastName: USER.lastName,
+                    email: USER.email
+                }
+                return util.successStatus(res, 201, 'User added successfully', data)
+            } catch(err){
+                return util.errorStatus(res, 500, 'Intername error');
```
Comment (chuxmykel) [inline server/controllers/userController.js]: Internal.

**Review by chuxmykel — CHANGES_REQUESTED** [review]

File: server/controllers/userController.js [hunk 6]
```diff
@@ -1,21 +1,17 @@
...
-              });
+          const user = await models.Users.findOne({where: {email: req.body.email}});
+            if(user){
+                //make changes to jude's work on naming convention change errorstatut to errorStatus
+                return util.errorStatus(res, 409, 'email address exist already');
             }
-
             const hashPassword = auth.hashPassword(req.body.password);
 
             const user = {
```
Comment (ocranbillions) [inline server/controllers/userController.js]: @OvieMudi @encodedBicoding Guys, pls check lines 10 and 17. const user is declared twice within the same scope

Comment (encodedBicoding) [reply server/controllers/userController.js]: This has been taken care of.

**Review by encodedBicoding — COMMENTED** [review]
Guys, please ignore the added file that had my work in it, I had to get the branch and work on it from my local machine, just so I could try to fix the merge conflict. that didn't work, I think Ovie has to be the one to fix that. Let me know if there are any minor changes to be made.

**Comment by chuxmykel** [comment]
@encodedBicoding Has the merge conflict been fixed?

**Pull request closed without merging**